src/lsm/manifest_level.zig

```diff
@@ -62,6 +131,8 @@ pub fn ManifestLevelType(
         keys: Keys,
         tables: Tables,
 
+        key_range: LevelKeyRange = .{ .key_range = null },
```

Comment:
@sentientwaffle: Ah, awesome catch dj! 🚀 

I had thought through a lot of scenarios related to persistent snapshots in the design phase of the ManifestLevel key ranges (specifically, how updating the `key_min/key_max` would work in [LevelKeyRange.exclude](https://github.com/tigerbeetle/tigerbeetle/blob/d763491a947bb6c3819d86f6cfb0e524b94e71d1/src/lsm/manifest_level.zig#L70-L99) if we were to consider persistent snapshots in the level). 

However, since we had started off using these key ranges only during negative lookups (which is the _write path_) -- I figured that we don't have to worry about persistent snapshots at all, we could ensure correctness by updating the key ranges only using the latest snapshot! But now, we're using the key ranges to optimize lookups in the _read path_ as well, so we would certainly have to account for persistent snapshots.

This is what I'm planning to do:
* Invoke the [exclusion](https://github.com/tigerbeetle/tigerbeetle/blob/d763491a947bb6c3819d86f6cfb0e524b94e71d1/src/lsm/manifest_level.zig#L217-L220) from the ManifestLevel key range in [ManifestLevel.remove_table](https://github.com/tigerbeetle/tigerbeetle/blob/d763491a947bb6c3819d86f6cfb0e524b94e71d1/src/lsm/manifest_level.zig#L256), instead of [ManifestLevel.set_snapshot_max](https://github.com/tigerbeetle/tigerbeetle/blob/d763491a947bb6c3819d86f6cfb0e524b94e71d1/src/lsm/manifest_level.zig#L204). This is because [ManifestLevel.remove_table] will be called for a snapshotted table when a persistent snapshot is being deleted, making it the right place to update the ManifestLevel's key range.
* In the [exclusion logic](https://github.com/tigerbeetle/tigerbeetle/blob/d763491a947bb6c3819d86f6cfb0e524b94e71d1/src/lsm/manifest_level.zig#L77-L98), would could directly iterate _level.tables_ in ascending/descending order (to find the next key_min/key_max)  instead of relying on an iterator over the tables visible to the latest snapshot, like we're doing right now.

Reply:
However,  a caveat while finding the next key_min (in case the table being removed contributes to the current key_min) is that we would have to iterate through all tables in the level to find out the next key_min. 😕 

This is because there could be cases where there exist two tables T & T' with key ranges [K-min, K-max] and [K'-min, K'-max], such that _K-max < K'-max, but K'-min < K-min_.

Reply:
:thinking: I see, we index on the key_max, so that is easy. But key_min.
I'll give this some more thought, but one possibility is just don't track the key-min at all -- only handle monotonically-increasing ids, not decreasing.

Reply:
Mhmmmm, but in that case, aren't we limiting our optimization to the negative lookup again? Maintaining the correct range would help positive lookups as well...

I'll think about this a little bit more too! For now, I'll implement the plan that I mentioned in https://github.com/tigerbeetle/tigerbeetle/pull/944#discussion_r1261458060  and measure performance with the same.

Reply:
Implemented this as a part of https://github.com/tigerbeetle/tigerbeetle/pull/944/commits/5ff1447d67368575e5f1e3b4b4b823b316bd2354. 

I'll think a bit harder about if there's a way for us to avoid iterating the entire level. I did the same before (during the design phase), but I hadn't been able to exhaustively think about all the cases required to come up with a more optimal solution. The current solution definitely has the potential to cause latency spikes during compaction, since [ManifestLevel.remove_table](https://github.com/tigerbeetle/tigerbeetle/blob/d763491a947bb6c3819d86f6cfb0e524b94e71d1/src/lsm/manifest_level.zig#L256) on a table that is contributing to the `key_min` of one of the lower levels of the LSM tree. 😕

Reply:
See https://github.com/tigerbeetle/tigerbeetle/pull/944#issuecomment-1642816324.